velox/dwio/parquet/tests/writer/ParquetWriterTest.cpp

```diff
@@ -146,6 +147,112 @@ TEST_F(ParquetWriterTest, compression) {
   assertReadWithReaderAndExpected(schema, *rowReader, data, *leafPool_);
 };
 
+TEST_F(ParquetWriterTest, updateDataPageVersion) {
+  auto schema = ROW({"c0"}, {INTEGER()});
+  const int64_t kRows = 1;
+  const auto data = makeRowVector({
+      makeFlatVector<int32_t>(kRows, [](auto row) { return 987; }),
+  });
+
+  // Write Parquet test data, then read and return the DataPage
+  // (thrift::PageType::type) used.
+  const auto testDataPageVersion = [&](parquet::WriterOptions writerOptions) {
+    // Create an in-memory writer.
+    auto sink = std::make_unique<MemorySink>(
+        200 * 1024 * 1024,
+        dwio::common::FileSink::Options{.pool = leafPool_.get()});
+    auto sinkPtr = sink.get();
+    writerOptions.memoryPool = leafPool_.get();
+
+    auto writer = std::make_unique<parquet::Writer>(
+        std::move(sink), writerOptions, rootPool_, schema);
+    writer->write(data);
+    writer->close();
+
+    dwio::common::ReaderOptions readerOptions{leafPool_.get()};
+    auto reader = createReaderInMemory(*sinkPtr, readerOptions);
+
+    auto colChunkPtr = reader->fileMetaData().rowGroup(0).columnChunk(0);
+    std::string_view sinkData(sinkPtr->data(), sinkPtr->size());
+
+    auto readFile = std::make_shared<InMemoryReadFile>(sinkData);
+    auto file = std::make_shared<ReadFileInputStream>(std::move(readFile));
+
+    auto inputStream = std::make_unique<SeekableFileInputStream>(
+        std::move(file),
+        colChunkPtr.dataPageOffset(),
+        150,
+        *leafPool_,
+        LogType::TEST);
+    auto pageReader = std::make_unique<PageReader>(
+        std::move(inputStream),
+        *leafPool_,
+        colChunkPtr.compression(),
+        colChunkPtr.totalCompressedSize());
+
+    return pageReader->readPageHeader().type;
+  };
+
+  // Test default behavior - DataPage should be V1.
+  config::ConfigBase connectorConfigDefault({});
+  config::ConfigBase connectorSessionPropertiesDefault({});
+
+  parquet::WriterOptions writerOptions;
+  writerOptions.processConfigs(
+      connectorConfigDefault, connectorSessionPropertiesDefault);
+
+  ASSERT_EQ(
+      testDataPageVersion(writerOptions), thrift::PageType::type::DATA_PAGE);
+
+  // Simulate setting DataPage version to V2, via Hive config from file.
+  std::unordered_map<std::string, std::string> configFromFile = {
+      {parquet::WriterOptions::kParquetHiveConnectorDataPageVersion, "V2"}};
+
+  config::ConfigBase connectorConfigV2(std::move(configFromFile));
+  writerOptions.processConfigs(
```

Comment:
processConfigs can go inside `testDataPageVersion`. This lambda will then take two arguments, session and config parameters.

Reply:
Fair, consolidated all repeated statements into the lambda.